packages/adapter-kysely/src/index.ts

```diff
@@ -0,0 +1,435 @@
+ *
+ * This adapter also exports a wrapper around the original Kysely class, `AuthedKysely`, that can be used to provide an additional level of type-safety. While using it isn't required, it is recommended as it will verify that the database interface has all the fields that Auth.js requires.
+ *
+ * ```ts title="db/index.ts"
+ * import type { GeneratedAlways } from "kysely"
+ * import { PostgresDialect } from "kysely"
+ * import { Pool } from "pg"
+ * import { AuthedKysely } from "@next-auth/kysely-adapter"
+ *
+ * interface User {
+ *   id: GeneratedAlways<string>
+ *   name: string | null
+ *   email: string
+ *   emailVerified: Date | null
+ *   image: string | null
+ * }
+ *
+ * interface Account {
+ *   id: GeneratedAlways<string>
+ *   userId: string
+ *   type: string
+ *   provider: string
+ *   providerAccountId: string
+ *   refresh_token: string | null
+ *   access_token: string | null
+ *   expires_at: number | null
+ *   token_type: string | null
+ *   scope: string | null
+ *   id_token: string | null
+ *   session_state: string | null
+ *   oauth_token_secret: string | null
+ *   oauth_token: string | null
+ * }
+ *
+ * interface Session {
+ *   id: GeneratedAlways<string>
+ *   userId: string
+ *   sessionToken: string
+ *   expires: Date
+ * }
+ *
+ * interface VerificationToken {
+ *   identifier: string
+ *   token: string
+ *   expires: Date
+ * }
+ *
+ * interface Database {
+ *   User: User
+ *   Account: Account
+ *   Session: Session
+ *   VerificationToken: VerificationToken
+ * }
+ *
+ * export const db = new AuthedKysely<Database>({
+ *   dialect: new PostgresDialect({
+ *     pool: new Pool({
+ *       host: process.env.DATABASE_HOST,
+ *       database: process.env.DATABASE_NAME,
+ *       user: process.env.DATABASE_USER,
+ *       password: process.env.DATABASE_PASSWORD,
+ *     }),
+ *   }),
+ * })
+```
+ *
+ *
+ * :::note
+ * An alternative to manually defining types is generating them from the database schema using [kysely-codegen](https://github.com/RobinBlomberg/kysely-codegen), or from Prisma schemas using [prisma-kysely](https://github.com/valtyr/prisma-kysely). When using generated types with `AuthedKysely`, import `Codegen` and pass it as the second generic arg:
+ * ```ts
+ * import type { Codegen } from "@next-auth/kysely-adapter"
+ * new AuthedKysely<Database, Codegen>(...)
+ * ```
+ * :::
+ * ### Schema
+ * ```ts title="db/migrations/001_create_db.ts"
+ * import { Kysely, sql } from "kysely"
+ *
+ * export async function up(db: Kysely<any>): Promise<void> {
+ *   await db.schema
+ *     .createTable("User")
+ *     .addColumn("id", "uuid", (col) =>
+ *       col.primaryKey().defaultTo(sql`gen_random_uuid()`)
+ *     )
+ *     .addColumn("name", "text")
+ *     .addColumn("email", "text", (col) => col.unique().notNull())
+ *     .addColumn("emailVerified", "timestamptz", (col) =>
+ *       col.defaultTo(sql`NOW()`)
+ *     )
```

Comment:
```suggestion
 *     .addColumn("emailVerified", "timestamptz")
```

I think this should be null by default instead.